### DISCUSSION TITLE
no matter what I do.

### DISCUSSION THREAD

**Discussion by DariusArya** [umputun/remark42#1131] [Q&A] [answered]
no matter what I do, I can't get remark42 to work on my modest website. 
even the demo page doesn't work. 
I want to get it to work on port 3031 because my port 8080 is taken by wildfly, 
but it issues the following errors:
GET: http://127.0.0.1:8080/web/embed.mjs
[HTTP/1.1 404  2ms]

GET: http://127.0.0.1:8080/web/counter.mjs
[HTTP/1.1 404  2ms]

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://127.0.0.1:8080/web/embed.mjs. (Reason: CORS request did not succeed).

Module source URI is not allowed in this document: “http://127.0.0.1:8080/web/embed.mjs”. web:1:1
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://127.0.0.1:8080/web/counter.mjs. (Reason: CORS request did not succeed).

Module source URI is not allowed in this document: “http://127.0.0.1:8080/web/counter.mjs”. web:1:1
Loading module from “http://127.0.0.1:8080/web/embed.mjs” was blocked because of a disallowed MIME type (“text/html”).
web
Loading module from “http://127.0.0.1:8080/web/counter.mjs” was blocked because of a disallowed MIME type (“text/html”).
web
GEThttps://comments.in-memori.am/favicon.ico
[HTTP/1.1 404 Not Found 0ms]

GEThttp://127.0.0.1:8080/web/last-comments.mjs
[HTTP/1.1 404  1ms]

Loading module from “http://127.0.0.1:8080/web/last-comments.mjs” was blocked because of a disallowed MIME type (“text/html”).
last-comments.html
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://127.0.0.1:8080/web/last-comments.mjs. (Reason: CORS request did not succeed).

Module source URI is not allowed in this document: “http://127.0.0.1:8080/web/last-comments.mjs”. last-comments.html:1:1

​I am running it on Ubuntu Linux, behind reverse proxy server nginx and on https, 
I have been spending hours on this, and I have followed all the instructions and documents I could find. 

What am I suppose to do ? 
Someone help me please ?

**Comment by umputun** [comment]
your query shows :8080 but you said " want to get it to work on port 3031". I don't know how exactly you install it, what version what config, and how all of this related to :8080

If you can post your configuration someone will be able to assist but without it, the only thing I can suggest is to change `host` value on the frontend side to whatever URL:port you are using and make sure the same set as `REMARK_URL` on the backend side.

**Reply by DariusArya** [reply]
I have installed 1.8, without docker, and just ran the binary in Ubuntu, 
  ./remark42.linux-amd64 server --port=3031 --secret=HuMSfURCpj --url="https://comments.in-memori.am" --auth.google.cid=somecode.apps.googleusercontent.com --auth.anon --auth.google.csec=googlesecret --site=in-memori.am

 I registered a subdomain for my domain, 
 and my nginx configuration for the subdomain is : 
 server {
    listen   443;
    listen [::]:443 ssl;
    root /var/www/remark42;
    index index.html;
    server_name comments.in-memori.am;

    ssl    on;
    ssl_certificate /etc/letsencrypt/live/in-memori.am-0001/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/in-memori.am-0001/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

    charset utf-8;

    location / {
      include conf.d/proxy_headers.conf;
      proxy_pass http://127.0.0.1:3031;
    }

}

server {
    listen 80;
    listen [::]:80;
    server_name comments.in-memori.am;
    return 301 https://$server_name$request_uri;
}

with headers as 
   proxy_set_header Host $host;
proxy_set_header X-Forwarded-Proto $scheme;
add_header Front-End-Https on;
add_header Cache-Control no-cache;
add_header Access-Control-Allow-Origin  always;

on the front end: 
I have tried various things, 

    var remark_config = {
    host: "https://comments.in-memori.am/", 
    site_id: 'in-memori.am',
    components: ['embed', 'last-comments', 'counter'], // optional param; which components to load. default to ["embed"]
                           // to load all components define components as ['embed', 'last-comments', 'counter']
                           // available component are:
                           //   - 'embed': basic comments widget
                           //   - 'last-comments': last comments widget, see `Last Comments` section below
                           //   - 'counter': counter widget, see `Counter` section below
    url: 'https://in-memori.am/wysiwyg/?name=U65U114U100U101U115U104U105U114U32U69U102U116U101U107U104U97U114U122U97U100U101U104', 
// optional param; if it isn't defined
                     // `window.location.origin + window.location.pathname` will be used
                     //
                     // Note that if you use query parameters as significant part of URL
                     // (the one that actually changes content on page)
                     // you will have to configure URL manually to keep query params, as
                     // `window.location.origin + window.location.pathname` doesn't contain query params and
                     // hash. For example, default URL for `https://example/com/example-post?id=1#hash`
                     // would be `https://example/com/example-post`
                     //
                     // The problem with query params is that they often contain useless params added by
                     // various trackers (utm params) and doesn't have defined order, so Remark42 treats differently
                     // all this examples:
                     // https://example.com/?postid=1&date=2007-02-11
                     // https://example.com/?date=2007-02-11&postid=1
                     // https://example.com/?date=2007-02-11&postid=1&utm_source=google
                     //
                     // If you deal with query parameters make sure you pass only significant part of it
                     // in well defined order
    max_shown_comments: 10, // optional param; if it isn't defined default value (15) will be used
   // theme: 'dark', // optional param; if it isn't defined default value ('light') will be used
    page_title: 'Moving to Remark42', // optional param; if it isn't defined `document.title` will be used
    locale: 'en', // set up locale and language, if it isn't defined default value ('en') will be used
    show_email_subscription: true, // optional param; by default it is `true` and you can see email subscription feature
                                    // in interface when enable it from backend side
                                    // if you set this param in `false` you will get notifications email notifications as admin
                                    // but your users won't have interface for subscription
    simple_view: true // optional param; overrides the parameter from the backend
                       // minimized UI with basic info only
  };
</script>
<script type="text/javascript">
!function(e,n){for(var o=0;o<e.length;o++){var r=n.createElement("script"),c=".js",d=n.head||n.body;"noModule"in r?(r.type="module",c=".mjs"):r.async=!0,r.defer=!0,r.src=remark_config.host+"/web/"+e[o]+c,d.appendChild(r)}}(remark_config.components||["embed"],document);
</script>   
with url left as default, and url: 'http://in-memori.am.' as other things I have tried. 

This should work, because in my nginx configuration, host: https://comments.in-memori.am should be redirected to http://127.0.0.1:3031
No ? 

But no matter what I do,  <div id=remark42></div> 
is empty and 
  https://comments.in-memori.am/web/ 
  does not bring out the editor or signin options, 
  
  Thank you for your help, I have been stuck in this for a long time,

**Comment by umputun** [comment, marked as answer]
if you load the page https://comments.in-memori.am/web/ and check web inspector you will see it hits http://127.0.0.1:8080/web. This is obviously wrong and likely has something to do with your frontend configuration. 

@akellbl4 any idea what's going on here?

_@DariusArya - I hope you posted fake secret and fake auth.google params here,  as this is a public forum_

**Reply by umputun** [reply]
@akellbl4 I just realized smth. How do we even handle the sed part for the URL replacements in such a standalone run mode (outside the docker)? 

Unless I'm missing something all we do in this mode is just serving static assets with embedded data without any adjustment. @paskal maybe you recall any details about it and if we had seen any working configuration like this?

To me, it seems like it won't work unless @DariusArya adjust links by running the same substitutions we do in [docker-init.sh](https://github.com/umputun/remark42/blob/master/docker-init.sh) and rebuild the binary. 

@DariusArya I would suggest using docker, if you don't want to deal with the rebuild. We are planning to change the substitution part and integrate it into the backend, but for now I think your only choices are to recompile your own binary or switch to the dockerized version

**Reply by akellbl4** [reply]
> @akellbl4 I just realized smth. How do we even handle the sed part for the URL replacements in such a standalone run mode (outside the docker)?

hm, I think we've done nothing in this case. Now the instance is down can't check it.

**Reply by DariusArya** [reply]
thank you, I think, I will use the dockerized version, though decovar.dev has been able to get the undockerized version to work.  
https://decovar.dev/blog/2021/05/30/new-server-and-remark42-comments/#remark42-comments
I tried to follow his example, but I think, I made a mistake somewhere, or may be he used a version that was different.
